streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -293,12 +301,11 @@ boolean tryToCompleteRestoration() {
      * @throws TaskMigratedException if the task producer got fenced (EOS only)
      */
     void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
-        final Set<TaskId> revokedTasks = new HashSet<>();
         final Set<TopicPartition> remainingPartitions = new HashSet<>(revokedPartitions);
 
         for (final Task task : tasks.values()) {
             if (remainingPartitions.containsAll(task.inputPartitions())) {
-                revokedTasks.add(task.id());
+                task.suspend();
```

Comment:
Simplify the logic here as we no longer throws before task suspend

Reply:
Not sure I follow: we can still throw from inside `suspend` right?

Reply:
What I mean is that we do not throw if remaining partitions are not empty anymore. We could directly suspend a task here.

Reply:
I see.